frontend/src/views/student/submissions/SubmissionView.vue

```diff
@@ -0,0 +1,307 @@
+      align: 'center'
+    }
+  ];
+
+  @Watch('editSubmissionDialog')
+  closeError() {
+    if (!this.editSubmissionDialog) {
+      this.currentSubmission = null;
+    }
+  }
+
+  async created() {
+    await this.getSubmissions();
+  }
+
+  async getSubmissions() {
+    await this.$store.dispatch('loading');
+    try {
+      [this.submissions, this.topics] = await Promise.all([
+        RemoteServices.getStudentSubmissions(),
+        RemoteServices.getTopics()
+      ]);
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+  }
+
+  customFilter(value: string, search: string, question: Question) {
+    // noinspection SuspiciousTypeOfGuard,SuspiciousTypeOfGuard
+    return (
+      search != null &&
+      JSON.stringify(question)
+        .toLowerCase()
+        .indexOf(search.toLowerCase()) !== -1
+    );
+  }
+
+  getStatusColor(status: string) {
+    if (status === 'AVAILABLE') return 'green';
+    else if (status === 'DISABLED') return 'orange';
+    else if (status === 'REJECTED') return 'red';
+    else if (status === 'IN_REVISION') return 'yellow';
+    else if (status === 'IN_REVIEW') return 'blue';
+  }
+
+  submitQuestion() {
+    let question = new Question();
+    question.status = 'IN_REVISION';
+    this.currentSubmission = new Submission();
+    this.currentSubmission.courseExecutionId = this.$store.getters.getCurrentCourse.courseExecutionId;
+    this.currentSubmission.question = question;
+    this.editSubmissionDialog = true;
+  }
+
+  editSubmission(submission: Submission, e?: Event) {
+    if (e) e.preventDefault();
+    this.currentSubmission = submission;
+    this.editSubmissionDialog = true;
+  }
+
+  async onSaveSubmission() {
+    await this.$store.dispatch('loading');
+    try {
+      [this.submissions] = await Promise.all([
+        RemoteServices.getStudentSubmissions()
+      ]);
+    } catch (error) {
+      await this.$store.dispatch('error', error);
+    }
+    await this.$store.dispatch('clearLoading');
+    this.editSubmissionDialog = false;
+    this.currentSubmission = null;
+  }
+
+  onSubmissionChangedTopics(questionId: Number, changedTopics: Topic[]) {
+    let submission = this.submissions.find(
+      (submission: Submission) => submission.question.id == questionId
+    );
+    if (submission) {
+      submission.question.topics = changedTopics;
+    }
+  }
+
+  showSubmissionDialog(submission: Submission) {
+    this.currentSubmission = submission;
+    this.submissionDialog = true;
+  }
+
+  onCloseShowSubmissionDialog() {
+    this.submissionDialog = false;
+  }
+
+  async deleteSubmission(toDeleteSubmission: Submission) {
+    if (
+      toDeleteSubmission.id &&
+      confirm('Are you sure you want to delete this submission?')
+    ) {
+      try {
+        let [reviews] = await Promise.all([
+          RemoteServices.getSubmissionReviews(toDeleteSubmission.id!)
+        ]);
+        if (reviews.length > 0) {
```

Comment:
Should not be done this way, a request to delete is send to the backend and an error message is displayed if it is not possible